hercules-ci-api/src/Hercules/API/Agents.hs

```diff
@@ -52,11 +55,21 @@ data AgentsAPI auth f = AgentsAPI
       Get '[JSON] [AgentSession]
 
   , agentSessionCreate :: f :-
-      Summary "Create a new agent session." :>
+      Summary "DEPRECATED" :>
       Description "Authenticated using the agent token acquired through POST /accounts/:accountId/clusterJoinTokens" :>
       "agentSessions" :>
-      ReqBody '[JSON] CreateAgentSession :>
+      ReqBody '[JSON] CreateAgentSession.CreateAgentSession :>
+      auth :>
+      Post '[JSON] Text
+
+  , agentSessionCreate2 :: f :-
```

Comment:
I wonder if there's a cleaner way in servant. I'd be tempted to try alternative combinatorbut not sure how that affects documentation generation - would it show both?

Reply:
I've tried providing both endpoints at the same route, which didn't backtrack, so failed. (So that's the "alternative" combinator, hidden by servant generic)
Scare quotes are because it's technically just a product type, which also means it won't have the right meaning inside ReqBody for example

Reply:
Alright let's go with `V2` since using sums is painful with all the generation.

Reply:
I've extrapolated a bit. Description is in the commit message:

> To conform to new naming convention:
>  - suffix "_V2" etc for modules
>  - no suffixes inside suffixed modules
>  - suffix "V2" etc for terms and types
> 
> The naming convention for modules is slightly different to make
> clear that it doesn't define a type with the basename of the module.